weatherv3/www/js/controllers.js

```diff
@@ -0,0 +1,116 @@
+angular.module('starter.controllers', ['ionic'])
+
+.constant('FORECASTIO_KEY', '76f3ef0ca4c614a2ced71001eb970aa8')
+
+
+.controller('HomeCtrl', function($scope,$state,Weather,DataStore) {
+    //read default settings into scope
+    console.log('inside home');
+    $scope.city  = DataStore.city;
+    var latitude  =  DataStore.latitude;
+    var longitude = DataStore.longitude;
+
+    //some code found here: https://www.binpress.com/tutorial/a-simple-weather-forecast-app-with-ionic-framework-and-forecastio/107
+    Weather.getCurrentWeather(latitude,longitude).then(function(resp) {
+      var current = resp.data;
+      $scope.temperature = Math.round(current.currently.temperature) + "°F";
+      $scope.apparentTemperature = Math.round(current.currently.apparentTemperature) + "°F";
+      $scope.summary = current.currently.summary.toLowerCase();
+      $scope.tempDif = Math.abs(Math.round(current.currently.temperature) - Math.round(current.currently.apparentTemperature));
+      $scope.windSpeed = Math.round(current.currently.windSpeed);
+      var intensity = current.currently.precipIntensity;
+
+      $scope.precipIntensity = 0;
+      if (intensity == 0) {
+        $scope.precipIntensity = 0;
+      } else if (intensity < .002) {
+        $scope.precipIntensity = "very lightly";
+      } else if (intensity < .017) {
+        $scope.precipIntensity = "lightly";
+      } else if (intensity < .1) {
+        $scope.precipIntensity = "moderately hard";
+      } else {
+        $scope.precipIntensity = heavily;
+      }
+      $scope.precipProbability = current.currently.precipProbability;
```

Comment:
No, I was wrong, but I don't think it makes a difference because I don't display the precipType if the intensity is 0, so having an undefined variable makes no difference.